elytron/WFLY-14735-Management-FORM-authentication.adoc

```diff
@@ -0,0 +1,245 @@
+
+==== Management Interfaces
+
+On the management interfaces a new attribute `use-core-management-security-policy` will be added,
+this will be a boolean and when set to `true` the configuration defined on the `policy=security`
+resource will be used instead of the individual attributes on the management interface.
+
+==== Out Of Scope
+
+The RFE will not be making changes to the SASL authentication for native access, the out of the
+box configuration will still default to supporting the JBOSS_LOCAL_USER and DIGEST SASL mechanisms.
+
+This RFE will also not be changing the getting started experience adding the initial users to the
+server using the add-user utility and this RFE will not be changing the properties files used to
+contain the initial users.
+
+=== Nice-to-Have Requirements
+
+Should the management interfaces have TLS enabled by default, we may be able to also generate a 
+wildfly-config.xml for clients to use poining at a Keystore for the server's certificate.
+
+This could however be high risk breaking various clients.
+
+=== Non-Requirements
+
+The architecture chosen for this RFE will form the basis for future development which will not be covered by
+this RFE.
+
+It is envisaged that the form authentication provided by the `elytron-authentication` subsystem will be 
+enhanced further with capabilities such as:
+
+ * Support for MFA authentication using OTP such as HOTP / TOTP.
+ * Dynamic decisions based on username such as recognising a user is registered for SSO.
+ * Fido2 / WebAuth authentication with or without a username.
+ * Multi factor authentication using push notifications.
+ * Customisation of the look and feel of the form.
+ 
+In addition to using the form for the management interfaces it will be possible to use configurations
+to secure deployments making all of these options available.
+
+These additional features are broadly the reason for placing the initial configuration for the form 
+into a new subsystem.
+
+== Backwards Compatibility
+
+=== Default Configuration
+
+Prior to this change the default configuration for the management interfaces referenced 
+`*-authentication-factory` resources to provide the SASL and HTTP authentication which was 
+DIGEST based, this change updates  the configuration to use a security policy withing the 
+core-management subsystem as well as changing  the default authentication of the HTTP management
+interface to use FORM authentication.
+
+Unless we also enable TLS for the management interfaces clients which connect to the native 
+interface either  directly or via HTTP Upgrade will be unaffected as this will still be using 
+SASL authentication which will be unchanged by this RFE.
+
+=== Importing Existing Configuration
+
+Existing configuration will be imported "as-is" and will continue to use the previously 
+defined behaviour, users who want to add form authentication to the HTTP management interface will 
+need to manually add it themselves.
```

Comment:
The Hard Requirements section should have some discussion of what the behavior is in this case (or any case where the new behavior is not enabled.)